### ISSUE TITLE
pycuda builds error on Fedora 37

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#3631]
We now build xpra without cuda on Fedora 37 531d640bf5641724ea6fd6f82471719a974e2860 because of this.

**Comment by totaam** [comment]
```
rpmbuild -ba packaging/rpm/python3-pycuda.spec
(..)
In file included from /usr/include/python3.11/Python.h:95:
/usr/include/python3.11/ceval.h:36:43: note: declared here
   36 | Py_DEPRECATED(3.9) PyAPI_FUNC(PyObject *) PyEval_CallFunction(
      |                                           ^~~~~~~~~~~~~~~~~~~
In file included from bpl-subset/bpl_subset/libs/python/src/object/class.cpp:15:
bpl-subset/bpl_subset/boost/python/object/pickle_support.hpp: In static member function ‘static void pycudaboost::python::detail::pickle_suite_registration::register_(Class_&, ...)’:
bpl-subset/bpl_subset/boost/python/object/pickle_support.hpp:110:31: warning: typedef ‘error_type’ locally defined but not used [-Wunused-local-typedefs]
  110 |           Class_>::error_type error_type;
      |                               ^~~~~~~~~~
In file included from /usr/include/python3.11/Python.h:44:
bpl-subset/bpl_subset/libs/python/src/object/class.cpp: In function ‘PyObject* pycudaboost::python::objects::static_data()’:
/usr/include/python3.11/object.h:136:30: error: lvalue required as left operand of assignment
  136 | #  define Py_TYPE(ob) Py_TYPE(_PyObject_CAST(ob))
      |                       ~~~~~~~^~~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:211:11: note: in expansion of macro ‘Py_TYPE’
  211 |           Py_TYPE(&static_data_object) = &PyType_Type;
      |           ^~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp: In function ‘pycudaboost::python::type_handle pycudaboost::python::objects::class_metatype()’:
/usr/include/python3.11/object.h:136:30: error: lvalue required as left operand of assignment
  136 | #  define Py_TYPE(ob) Py_TYPE(_PyObject_CAST(ob))
      |                       ~~~~~~~^~~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:319:11: note: in expansion of macro ‘Py_TYPE’
  319 |           Py_TYPE(&class_metatype_object) = &PyType_Type;
      |           ^~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp: In function ‘PyObject* pycudaboost::python::objects::instance_new(PyTypeObject*, PyObject*, PyObject*)’:
/usr/include/python3.11/object.h:145:30: error: lvalue required as left operand of assignment
  145 | #  define Py_SIZE(ob) Py_SIZE(_PyObject_CAST(ob))
      |                       ~~~~~~~^~~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:378:15: note: in expansion of macro ‘Py_SIZE’
  378 |               Py_SIZE(result) =
      |               ^~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp: In function ‘pycudaboost::python::type_handle pycudaboost::python::objects::class_type()’:
/usr/include/python3.11/object.h:136:30: error: lvalue required as left operand of assignment
  136 | #  define Py_TYPE(ob) Py_TYPE(_PyObject_CAST(ob))
      |                       ~~~~~~~^~~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:473:11: note: in expansion of macro ‘Py_TYPE’
  473 |           Py_TYPE(&class_type_object) = incref(class_metatype().get());
      |           ^~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp: In static member function ‘static void* pycudaboost::python::instance_holder::allocate(PyObject*, std::size_t, std::size_t)’:
/usr/include/python3.11/object.h:145:30: error: lvalue required as left operand of assignment
  145 | #  define Py_SIZE(ob) Py_SIZE(_PyObject_CAST(ob))
      |                       ~~~~~~~^~~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:742:9: note: in expansion of macro ‘Py_SIZE’
  742 |         Py_SIZE(self) = holder_offset;
      |         ^~~~~~~
In file included from bpl-subset/bpl_subset/pycudaboost/preprocessor/iteration/detail/iter/forward1.hpp:62:
bpl-subset/bpl_subset/pycudaboost/python/call.hpp: In instantiation of ‘typename pycudaboost::python::detail::returnable<T>::type pycudaboost::python::call(PyObject*, const A0&, const A1&, const A2&, pycudaboost::type<T>*) [with R = api::object; A0 = const char*; A1 = handle<>; A2 = dict; typename detail::returnable<T>::type = api::object; PyObject = _object]’:
bpl-subset/bpl_subset/pycudaboost/python/object_call.hpp:19:25:   required from ‘typename pycudaboost::python::detail::dependent<pycudaboost::python::api::object, A0>::type pycudaboost::python::api::object_operators<U>::operator()(const A0&, const A1&, const A2&) const [with A0 = const char*; A1 = pycudaboost::python::handle<>; A2 = pycudaboost::python::dict; U = pycudaboost::python::api::object; typename pycudaboost::python::detail::dependent<pycudaboost::python::api::object, A0>::type = pycudaboost::python::api::object]’
bpl-subset/bpl_subset/libs/python/src/object/class.cpp:575:47:   required from here
bpl-subset/bpl_subset/pycudaboost/python/call.hpp:62:28: warning: ‘PyObject* PyEval_CallFunction(PyObject*, const char*, ...)’ is deprecated [-Wdeprecated-declarations]
   62 |     PyObject* const result =
      |         ~~~~~~~~~~~~~~~~~~~^
   63 |         PyEval_CallFunction(
      |         ~~~~~~~~~~~~~~~~~~~~
   64 |             callable
      |             ~~~~~~~~        
   65 |             , const_cast<char*>("(" BOOST_PP_REPEAT_1ST(N, BOOST_PYTHON_FIXED, "O") ")")
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   66 |             BOOST_PP_REPEAT_1ST(N, BOOST_PYTHON_FAST_ARG_TO_PYTHON_GET, nil)
      |             ~               
/usr/include/python3.11/ceval.h:36:43: note: declared here
   36 | Py_DEPRECATED(3.9) PyAPI_FUNC(PyObject *) PyEval_CallFunction(
      |                                           ^~~~~~~~~~~~~~~~~~~
bpl-subset/bpl_subset/pycudaboost/python/call.hpp:62:28: warning: ‘PyObject* PyEval_CallFunction(PyObject*, const char*, ...)’ is deprecated [-Wdeprecated-declarations]
   62 |     PyObject* const result =
      |         ~~~~~~~~~~~~~~~~~~~^
   63 |         PyEval_CallFunction(
      |         ~~~~~~~~~~~~~~~~~~~~
   64 |             callable
      |             ~~~~~~~~        
   65 |             , const_cast<char*>("(" BOOST_PP_REPEAT_1ST(N, BOOST_PYTHON_FIXED, "O") ")")
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   66 |             BOOST_PP_REPEAT_1ST(N, BOOST_PYTHON_FAST_ARG_TO_PYTHON_GET, nil)
      |             ~               
/usr/include/python3.11/ceval.h:36:43: note: declared here
   36 | Py_DEPRECATED(3.9) PyAPI_FUNC(PyObject *) PyEval_CallFunction(
      |                                           ^~~~~~~~~~~~~~~~~~~
error: command '/usr/bin/gcc' failed with exit code 1
error: Bad exit status from /var/tmp/rpm-tmp.HQ52k5 (%build)

RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.HQ52k5 (%build)
```

**Comment by iDmple** [comment]
Hello,
I'm having this problem building the master branch on Ubuntu 20.04 regarding cuda:
```
make[1]: Entering directory '/xpra'                                                                                                  
dh_movefiles                                                                                                                         
dh_movefiles: warning: debian/tmp/usr/share/xpra/cuda not found (supposed to put it in xpra-codecs-nvidia)                           
tar: /xpra/debian/movelist: Cannot stat: No such file or directory                                                                   
tar: Error is not recoverable: exiting now                                                                                           
tar: This does not look like a tar archive                                                                                           
tar: Exiting with failure status due to previous errors                                                                              
dh_movefiles: error: (cd debian/tmp >/dev/null ; tar --create --files-from=/xpra/debian/movelist --file -) | (cd debian/xpra-codecs-nvidia >/dev/null ;tar xpf -) returned exit code 2                                                                                    
make[1]: *** [debian/rules:22: override_dh_install] Error 25                                                                         
make[1]: Leaving directory '/xpra'                                                                                                   
make: *** [debian/rules:12: binary] Error 2                                                                                          
dpkg-buildpackage: error: debian/rules binary subprocess returned exit status 2                                                      
debuild: fatal error at line 1182:                                                                                                   
dpkg-buildpackage -us -uc -ui -b failed
```

Is it related to this issue? This is a new problem that started recently. Thanks!

**Comment by totaam** [comment]
@iDmple your problem is not related to pycuda or Fedora 37, please don't use this ticket.
As for the cause, you don't have `cuda` installed so you cannot build the `xpra-codecs-nvidia` package.
Either remove it from the `control` and `rules` files or install the dependencies needed for building the `nvidia` bits.

**Comment by totaam** [comment]
Fixed in 8cec239aaf30741d1ca55e4dbb0b469b1b9c45c6 + cbb0b3aca755a31c75b28ad507a056559327ce81
